Mejoras en las tarjetas de pedido/carrito

DanyElAlgo opened the pull request:
Ahora son más configurables, se pueden mostrar contenido deseado mediante un mockup que está en el mismo archivo. Aún se debe hacer que se traiga el contenido desde supabase

DanyElAlgo committed:
Updating

DanyElAlgo committed:
mejoras en las tarjetas de orden y carrito

todavía se debe implementar supabase :c

DanyElAlgo committed:
Merge remote-tracking branch 'upstream/main'

Jonath117 commented on OrderNow-React/src/App.jsx:
```diff
@@ -9,15 +9,16 @@ import Cart from './pages/Cart.jsx'
+        <Layout>
+          <Routes>
+            <Route path='/' element={ <Home/> }></Route>
+            <Route path="/restaurantes" element={<Restaurantes />} />
```
faltó importar Restaurantes  o tiene otro nombre?, lo demás está bien, pero con esa línea no me carga lo demás

DanyElAlgo replied:
Fue cambiado por alguien más, se debe renombrar a businesses

Jonath117 reviewed (commented):
Revisar PR de mi partner

DanyElAlgo committed:
Merge remote-tracking branch 'upstream/main'

DanyElAlgo committed:
creación de checkout

al igual que el carrito, aún está tomando valores predefinidos en vez de usar un sistema de guardado local

DanyElAlgo committed:
cambios en el checkout

ahora es una tabla apropiada, aún falta hacer GET y POST

RaidenMT commented on OrderNow-React/src/components/OrderCard.jsx:
```diff
@@ -1,46 +1,91 @@
 
-function OrderCard(props) {
- 
+function OrderCard({
```
Si es un componente hacerlo constante y al final exportarlo, como los otros componentes

RaidenMT commented on OrderNow-React/src/pages/Cart.jsx:
```diff
@@ -1,18 +1,110 @@
+import React from "react";
+import { useState } from "react";
+
+//! SOLO UTILIZAR ENLACES DE IMÁGENES, NO USAR FOTOS LOCALES!!!!
```
Quitar comentario

RaidenMT commented on OrderNow-React/src/pages/Cart.jsx:
```diff
@@ -1,18 +1,110 @@
+
+//! SOLO UTILIZAR ENLACES DE IMÁGENES, NO USAR FOTOS LOCALES!!!!
+
+const fakeProducts = [
```
Pasarlo a archivo json y leerlo desde ahí. Jean Paul creó una carpeta de datos falsos. Usar la misma carpeta

RaidenMT commented on OrderNow-React/src/pages/Cart.jsx:
```diff
@@ -1,18 +1,110 @@
+  }
+
+  const navigate = useNavigate();
+  const restaurant = () => {
```
llamarlo goToBusiness

RaidenMT commented on OrderNow-React/src/pages/Cart.jsx:
```diff
@@ -1,18 +1,110 @@
+  const restaurant = () => {
+    navigate(`/restaurantes`);
+  };
+  const checkout = () => {
```
Llamarlo goToCheckout

RaidenMT commented on OrderNow-React/src/pages/Checkout.jsx:
```diff
@@ -0,0 +1,86 @@
+import { useState } from "react";
+import CheckoutElement from "../components/checkout-element/checkout-element";
+
+const fakeProducts = [
```
Pasar a archivo json

RaidenMT commented on OrderNow-React/src/pages/Checkout.jsx:
```diff
@@ -0,0 +1,86 @@
+  0
+);
+
+function Checkout() {
```
Poner const

RaidenMT commented on OrderNow-React/src/pages/Checkout.jsx:
```diff
@@ -0,0 +1,86 @@
+  const [products, setProducts] = useState(fakeProducts);
+  const [price, setPrice] = useState(totalPrice);
+  const navigate = useNavigate();
+  const restaurant = () => {
```
Duplicado en la otra página. Si es necesario crear un nuevo archivo/clase para gestionar las rutas hacerlo para no tener en diferentes páginas las mismas navegaciones de páginas

RaidenMT commented on OrderNow-React/src/pages/Restaurante.jsx:
¿Quién trabajó esto? ¿No es de otro equipo? Si fue de ejemplo para poder trabajar el Cart explicar

DanyElAlgo replied:
Esa página no me pertenece, es de otro grupo, de hecho, en ningún momento debí tocar esta página, solo trabajé con Cart y Checkout

RaidenMT reviewed (changes requested):
Revisar primero estos comentarios antes de que yo pruebe las funcionalidades por favor

DanyElAlgo committed:
uso de context para guardar información

El detalle está en que aún no logro actualizar el contexto en la sección del carrito. Pero lo delegaré por el momento, debo ver ahora el checkout

DanyElAlgo committed:
Merge remote-tracking branch 'upstream/main'

DanyElAlgo committed:
aún no funciona el ajuste del carrito

he intentado usar un nuevo método y ya me deja listar los productos y editar sus datos, incluso eliminarlos, el problema es que ahora no se actualizan los datos al pasarlos al checkout

DanyElAlgo committed:
Gestión de carrito completada

el carrito funciona a la perfección, ahora se puede mandar sin problemas los datos del carrito al checkout incluso cuando se editaron los elementos

Aunque aún tiene dos detalles menores, que al actualizar la página se reinicia el context y que aún no obtiene datos de la base de datos

RaidenMT commented on OrderNow-React/src/context/CartContext.jsx:
```diff
@@ -0,0 +1,91 @@
+  }
+}
+
+const fakeProducts = [
```
Agregar en un archivo de fakeData, basarse en los otros que ya existen

RaidenMT commented on OrderNow-React/src/pages/Cart.jsx:
```diff
@@ -2,109 +2,97 @@ import OrderCard from "../components/OrderCard";
+    <cartListContext value={products}>
+      <div className="space-y-4 flex flex-col items-center mt-3 pt-24">
+        {products.map((product) => (
+          <OrderCard
```
![image](https://github.com/user-attachments/assets/6cef972c-3775-48ac-90b9-79997f18b1a9)

Al entrar por computadora, no se ven el botón al final

DanyElAlgo replied:
Este problema se está viendo en otro PR, no es parte mía

RaidenMT commented on OrderNow-React/src/pages/Checkout.jsx:
```diff
@@ -1,86 +1,61 @@
       <div className="flex flex-col items-center">
         <h3 className="text-xl m-2">Total del pedido: {price} Bs.</h3>
-        <div onClick={restaurant}>
+        <div onClick={goToBusiness}>
```
Que lo mande al home, no a restaurantes

RaidenMT reviewed (changes requested):
Ya son los últimos cambios, revisar por favor

DanyElAlgo committed:
Merge remote-tracking branch 'upstream/main'

DanyElAlgo committed:
Mock data fuera del contexto

ahora los datos se declaran en un archivo aparte, y de ahí se importan al context
